pkg/skaffold/runner/v1/new.go

```diff
@@ -89,8 +89,11 @@ func NewForConfig(runCtx *runcontext.RunContext) (*SkaffoldRunner, error) {
 		return nil, fmt.Errorf("creating tester: %w", err)
 	}
 	syncer := getSyncer(runCtx)
+
+	podSelector := kubernetes.NewImageList()
+
 	var deployer deploy.Deployer
-	deployer, err = getDeployer(runCtx, labeller.Labels())
+	deployer, err = getDeployer(runCtx, podSelector, labeller.Labels())
```

Comment:
Since this is an empty `ImageList` can we just create a new `podSelector` in the individual deployer?

Reply:
well, this `podSelector` is actually being shared across all `Deployer`s and added to across runs. however, working on the next PR (actually moving the `Logger` into the `Deployer`) I realized that this won't work, because we need to have individual `podSelector`s in the `Deployer`s, but we also need to keep references to them in the `Runner` for the other components that haven't been moved in the interim.

I think the cleanest solution is to temporarily return the `podSelector` from `NewDeployer` so we can track references in the `Runner`, and remove this once all three `podSelector`-dependent components have been moved out of the `Runner`. I added this in a follow up commit here - let me know what you think about it

Reply:
yeah. After i wrote the comment, i realized we still need the reference to this podSelector